internal/pkg/cli/task_run.go

```diff
@@ -377,6 +391,72 @@ func (o *runTaskOpts) Validate() error {
 		}
 	}
 
+	for _, value := range o.secrets {
+		//For Systems Manager Parameter Store parameter you can specify it as name or the ARN if it exists in the same Region as the task you are launching.
+		if !isSSM(value) && !isSecretsManager(value) {
+			return fmt.Errorf("must specify a valid secrets ARN")
+		}
+	}
+
+	return nil
+}
+
+func isSSM(value string) bool {
+	// For SSM parameter you can specify it as ARN or name if it exists in the same Region as the task you are launching.
+	return !template.IsARNFunc(value) || strings.Contains(value, ":ssm:")
+}
+
+func isSecretsManager(value string) bool {
+	return strings.Contains(value, ":secretsmanager:")
+}
+
+func (o *runTaskOpts) categorizeSecrets() {
+	o.secretsManagerSecrets = make(map[string]string)
+	o.ssmParamSecrets = make(map[string]string)
```

Comment:
The reason we are initializing them here is because they will be used only in case customers doesn't specify app and env and then we will initiate them and use it further, else we will not initiate them.

Reply:
No they are used with or without app and env right? Like as long as there is `--secrets` they are categorized into either ssm or secretsmanager.

Reply:
Oops sorry, that was the case before not now. Yo are right Wan